robottelo/cli/base.py

```diff
@@ -431,7 +433,7 @@ def set_parameter(cls, options=None):
         return result
 
     @classmethod
-    def update(cls, options=None):
+    def update(cls, options=None, user=None, password=None):
```

Comment:
We can think in a solution that keeps the usability and compatibility of `make_user` classmethod (I really like that style). But also avoiding race conditions, it can be done by creating a new instance inside `make_user` method and dealing with instance attributes, or by creating a deepcopy of the class.

If we are going to rethink all `cli` architecture better to do that in parallel in a separate project/repo as @omaciel mentioned the idea is to have another `*guns` for UI and CLI. 

BTW: lets call it `candygun` or `flowergun` or `cotton-hammer` hehe to get rid of all the violence. :P